### PR TITLE
Ladybird: Add bookmarks bar

### PR DESCRIPTION
**Pull request by guerinoni**
Now it is possible to add a website to bookmarks bar with the plus button
near the url and with right click is possible to delete it one.

https://user-images.githubusercontent.com/41150432/230882553-9f158409-f432-419b-9e75-5fd50e4388e0.mov

### CODE REVIEW

File: Ladybird/Tab.cpp [hunk 1]
```diff
@@ -247,6 +264,49 @@ int Tab::tab_index()
...
+    if (bm->get_urls().isEmpty())
+        return;
+
+    auto idx = list_view->indexAt(pos);
+
+    QMenu contextMenu("Context menu", this);
+    QAction remove_action("Remove", this);
+    QObject::connect(&remove_action, &QAction::triggered, this, [bm, idx] { bm->remove(idx.row()); });
+
+    contextMenu.addAction(&remove_action);
```
Comment (MacDue) [inline Ladybird/Tab.cpp:306]: It looks like the `remove_action` will go out of scope after this function? Is it safe to pass a pointer to it?

File: Ladybird/Tab.cpp [hunk 2]
```diff
@@ -30,6 +32,15 @@ Tab::Tab(BrowserWindow* window, StringView webdriver_content_ipc_path, WebView::
 
     m_view = new WebContentView(webdriver_content_ipc_path, enable_callgrind_profiling);
     m_toolbar = new QToolBar(this);
+    m_bookmarks_bar = new QListView(this);
+    m_bookmarks_bar->setFlow(QListView::LeftToRight);
```
Comment (MacDue) [inline Ladybird/Tab.cpp:36]: Feels a little odd each tab has its own bookmark bar, but maybe okay :)

File: Ladybird/BrowserWindow.cpp [hunk 3]
```diff
@@ -337,7 +338,7 @@ BrowserWindow::BrowserWindow(Browser::CookieJar& cookie_jar, StringView webdrive
 void BrowserWindow::set_current_tab(Tab* tab)
 {
     m_current_tab = tab;
-    if (tab)
+    if (m_current_tab)
         update_displayed_zoom_level();
 }
```
Comment (MacDue) [inline Ladybird/BrowserWindow.cpp:343]: Unrelated change

File: Ladybird/Tab.cpp (hunk 2 above)
Comment (guerinoni) [reply Ladybird/Tab.cpp:36]: Yes, but the model is just one... Right now for every tab all widget are "duplicated"

File: Ladybird/BrowserWindow.cpp (hunk 3 above)
Comment (guerinoni) [reply Ladybird/BrowserWindow.cpp:343]: Fixed

File: Ladybird/Tab.cpp (hunk 1 above)
Comment (guerinoni) [reply Ladybird/Tab.cpp:306]: It's safe because it is just the "moment" when context menu appears, if you click outside make sense all stuff are deallocated

**Comment by AtkinsSJ** [comment]
This has conflicts.

Just going by the video, I feel like the bookmarks should look more like buttons? They don't seem to match very well.
